Src/Particle/AMReX_NeighborParticlesGPUImpl.H

```diff
@@ -221,25 +190,23 @@ buildNeighborCopyOp ()
         auto p_src_indices = neighbor_copy_op.m_src_indices[lev][gid].dataPtr();
         auto p_periodic_shift = neighbor_copy_op.m_periodic_shift[lev][gid].dataPtr();
 
+        Gpu::streamSynchronize();
         AMREX_FOR_1D ( np, i,
         {
-            IntVect iv = getParticleCell(p_ptr[i], plo, dxi, domain);
-            int code = mask_arr(iv);
-            if (code >= 0)
-            {
-                int k = 0;
-                int start = p_code_offsets[code];
-                int stop =  p_code_offsets[code+1];
-                for (int j = start; j < stop; ++j)
-                {
-                    int index = p_offsets[i]+k;
-                    p_boxes[index] = p_code_array[j].grid_id;
-                    p_levs[index] = 0;
-                    p_periodic_shift[index] = p_code_array[j].periodic_shift;
-                    p_src_indices[index] = i;
+            int pid = i;
+            if (use_boundary_neighbor)  pid = p_bndry_pid[i];
```

Comment:
Likewise here, please.

Reply:
Sure, I have changed them in the new commit. Thanks.